### PR TITLE
Build from ubi8 with CentOS 8-stream repos

### PR DESCRIPTION
**Pull request by Fryguy**
This also supports builds for other arches.

@bdunne Please review.  To more easily see the changes, only look at the 4 most recent commits: https://github.com/Fryguy/container-postgresql/compare/Fryguy:ubi8~4...Fryguy:ubi8

Part of https://github.com/ManageIQ/manageiq-appliance-build/issues/459

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ RUN cd /tmp && \
     if [[ "$(cat .git/HEAD)" == "ref:"* ]]; then sha=$(cat .git/$sha); fi && \
     echo "$(date +"%Y%m%d%H%M%S")-$sha" > /tmp/BUILD
 
-FROM centos/postgresql-10-centos8:latest
+FROM postgresql-container:10
```
Comment (Fryguy) [inline Dockerfile]: I think there's a nicer way to represent a "local" image, but I'm not sure what that would be.

File: bin/build_container_image [hunk 2]
```diff
@@ -2,13 +2,22 @@
 
 REGISTRY=${REGISTRY:-"docker.io"}
 ORGANIZATION=${ORGANIZATION:-"manageiq"}
+
+[[ -z "$TRAVIS_BRANCH" ]] && TRAVIS_BRANCH="$(git rev-parse --abbrev-ref HEAD)"
```
Comment (Fryguy) [inline bin/build_container_image:6]: Locally, without this, you get a build tagged `latest-`.  After this you get `latest-branch` when not on master.

File: bin/build_container_image [hunk 3]
```diff
@@ -2,13 +2,22 @@
...
 DEFAULT_TAG="10"
 [[ $TRAVIS_BRANCH != "master" ]] && DEFAULT_TAG="latest-$TRAVIS_BRANCH"
 TAG=${TAG:-"$DEFAULT_TAG"}
+
 IMAGE_NAME=$REGISTRY/$ORGANIZATION/postgresql:$TAG
 
 set -e
 
+rm -rf images/postgresql-container
+git clone --branch generated --depth 1 https://github.com/sclorg/postgresql-container images/postgresql-container
```
Comment (Fryguy) [inline bin/build_container_image]: Maybe better to just pull a tarball, but we have git here, so this seemed easy enough

**Comment by kbrock** [comment]
what does green mean here?
Does this mean that the container builds?

**Comment by Fryguy** [comment]
> what does green mean here?
> Does this mean that the container builds?

Yup: https://github.com/ManageIQ/container-postgresql/blob/178417aa764abe5119217d4a8a5e1625de642382/.travis.yml#L4

**Comment by bdunne** [comment]
What do you think of merging all of this into one `Dockerfile` with multiple layers?  We can do the git clone in a separate layer and copy in the files that are needed.  Then we don't need to tag the intermediate layer either.

**Comment by Fryguy** [comment]
The problem is that the docker build needs to be done from the clone itself...can you do that in an intermediate layer?

**Comment by bdunne** [comment]
I'm thinking it would be something like this:
```
# Just git resources you need
FROM uib:8 AS centos_postgresql
RUN dnf -y install git
RUN git clone ...


# Intermediate layer contents from your images/Dockerfile.ubi8 file.
FROM ubi:8 AS ubi_postgres
... your dockerfile contents here, just modify the COPY commands as required ...
COPY --from=centos_postgresql /source/location /destination


# Final image (our current Dockerfile)
FROM ubi_postgres
...
```
We do something similar [here](https://github.com/ManageIQ/manageiq-pods/blob/87d2c0b747a7fcd09961864bb99225baf42a0fe4/images/manageiq-base/Dockerfile#L63)

**Comment by Fryguy** [comment]
@bdunne Updated.

**Comment by Fryguy** [comment]
I looked into using a tarball and not git, but it turns out tar and gzip are also not in ubi8-minimal, so I'd need an install step anyway.  Up to you what you prefer

```diff
- RUN microdnf -y --setopt=tsflags=nodocs install git
- RUN git clone --branch generated --depth 1 https://github.com/sclorg/postgresql-container /postgresql-container
+ RUN microdnf -y --setopt=tsflags=nodocs install tar gzip
+ RUN mkdir postgresql-container && \
+     curl -L https://github.com/sclorg/postgresql-container/tarball/generated | tar xz --strip-components 1 -C postgresql-container
```

**Comment by Fryguy** [comment]
Also verified that I can rake db:create and db:migrate with this, so it works (notice I picked a different port, 5454, to ensure it was its own pg instance)

```
$ docker run --rm -it -p 5454:5432 -e POSTGRESQL_USER=root -e POSTGRESQL_PASSWORD=smartvm -e POSTGRESQL_DATABASE=vmdb_development docker.io/manageiq/postgresql:10

$ DATABASE_URL="postgres://root:smartvm@localhost:5454/vmdb_development" be rake db:create db:migrate
```

**Pull request merged**